### PR TITLE
Format doc comments

### PR DESCRIPTION
**Pull request by AldaronLau**
I find the lack of consistency in formatting in rust docs for core/std/alloc to be annoying.

This PR formats code blocks in doc comments by running the unstable `format_code_in_doc_comments`.

Running `cargo fmt` with this feature does make tidy fail (with trailing whitespace), so it had to be commented out in the rustfmt.toml after adding and running.

One thing I'm not sure about is why there were other formatting changes when I ran it.  If needed, I could try to go through them and only grab the doc comment changes for the core/std/alloc libraries, removing the rest of the changes.

### CODE REVIEW

File: compiler/rustc_apfloat/src/lib.rs [hunk 1]
```diff
@@ -538,7 +537,6 @@ pub trait Float:
     ///   NaN -> \c IEK_NAN
     ///   0   -> \c IEK_ZERO
     ///   Inf -> \c IEK_INF
-    ///
```
Comment (eddyb) [inline compiler/rustc_apfloat/src/lib.rs:541]: Dismissing my ping on the basis of these being trivial comment-only changes (tho I wish we could've just frozen this directory *sigh*).

> If needed, I could try to go through them and only grab the doc comment changes for the core/std/alloc libraries, removing the rest of the changes.

Consider this comment a (tiny) vote for having a PR just for `library/`, the compiler itself in general is more likely to be chaotic, and even have weird snippets in comments etc.

But if people are willing to review the PR as-is, I have nothing against that, to be clear.

File: compiler/rustc_const_eval/src/interpret/terminator.rs [hunk 2]
```diff
@@ -534,7 +534,7 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let receiver_place = loop {
                     match receiver.layout.ty.kind() {
                         ty::Ref(..) | ty::RawPtr(..) => break self.deref_operand(&receiver)?,
-                        ty::Dynamic(..) => break receiver.assert_mem_place(), // no immediate unsized values
+                        ty::Dynamic(..) => break receiver.assert_mem_place(), /* no immediate unsized values */
```
Comment (RalfJung) [inline compiler/rustc_const_eval/src/interpret/terminator.rs:537]: Some of these are strange -- there is no reason this needs changing I think

**Comment by RalfJung** [comment]
Generally we don't accept formatting PRs unless they are checked by tidy. Also this PR is way too big to be reviewable so it'd be advisable to split it up.

There's a question here of whether the time people will spend reviewing this (in particular for the part outside of `library`) couldn't be better spent on other work.

File: compiler/rustc_codegen_cranelift/src/base.rs [hunk 3]
```diff
@@ -501,7 +501,7 @@ fn codegen_stmt<'tcx>(
 
     #[cfg(any())] // This is never true
     match &stmt.kind {
-        StatementKind::StorageLive(..) | StatementKind::StorageDead(..) => {} // Those are not very useful
+        StatementKind::StorageLive(..) | StatementKind::StorageDead(..) => {} /* Those are not very useful */
```
[the remark this answers is not in the document]
Comment (AldaronLau) [reply compiler/rustc_codegen_cranelift/src/base.rs:504]: I don't understand why rust-fmt made that change.

I'll go through each change manually and remove changes like this, and just remove all changes to compiler/ anyway.

**Review by scottmcm — CHANGES_REQUESTED** [review]
I'm with Ralf on this one -- one PR to do this across everything is too big, and procedurally can't really happen because different people should review the different parts.

I was tempted to just close, but it sounds like you're going to use this one for just library changes, so that would be ok I think.  You might still consider closing this one and opening a new one so that you'll have a fresh PR without all the pings to the will-be-irrelevant people and teams.

**Comment by wesleywiser** [comment]
Hi @AldaronLau! First off, let me say "thank you" for spending the time and effort to help improve the consistency of the repo's doc comments. 🙂 

As @RalfJung mentions, reviewer time is both limited and precious and so for PRs like this is, it's critical that they be optimized for ease of review. To that end, I have a few suggestions that might help:

- Split the changes into sets of distinct changes. A quick look at the diff suggests at least a few categories of changes (remove empty trailing doc comment, add whitespace around inline comment contents, rustfmt of doc codeblocks, etc) each of which could be their own commits and PRs.
- Teach `tidy` to check for these issues. @RalfJung mentioned that we generally don't accept formatting PRs that aren't checked by tidy because it's hard to keep the formatting nice without tidy enforcing it and making it human enforced adds to reviewer load. 
- Try to minimize the scope of each individual PR (@scottmcm's suggestion is a good step towards this). That will require less time from each reviewer and also make it easier to merge the PRs themselves. (This PR has been open <24 hours and already has merge conflicts. Smaller PRs will be less likely to conflict with other PRs in the bors queue.)

Thanks for your help in this area!

**Comment by AldaronLau** [comment]
> Teach `tidy` to check for these issues. @RalfJung mentioned that we generally don't accept formatting PRs that aren't checked by tidy because it's hard to keep the formatting nice without tidy enforcing it and making it human enforced adds to reviewer load.

I'm not familiar with `tidy`, where can I get more information on how to "teach" it to enforce specific formatting rules?

**Comment by wesleywiser** [comment]
The `tidy` tool lives in [`src/tools/tidy`](https://github.com/rust-lang/rust/tree/master/src/tools/tidy) and you'll probably want to add some new checks in [`ui_tests.rs`](https://github.com/rust-lang/rust/blob/master/src/tools/tidy/src/ui_tests.rs) for things like "remove empty trailing doc comment" and "add whitespace around inline comment contents" sets of changes. With those two out of the way, perhaps enabling the `format_code_in_doc_comments` rustfmt option could be done a few crates at a time?

**Comment by JohnCSimon** [comment]
ping from triage:
@AldaronLau 

Can you please address the merge conflicts - and post your status on this PR?

FYI: when a PR is ready for review, send a message containing 
`@rustbot ready` to switch to `S-waiting-on-review` so the PR is in the reviewer's backlog.

**Comment by AldaronLau** [comment]
> ping from triage: @AldaronLau
> 
> Can you please address the merge conflicts - and post your status on this PR?
> 
> FYI: when a PR is ready for review, send a message containing `@rustbot ready` to switch to `S-waiting-on-review` so the PR is in the reviewer's backlog.

Sorry, I forgot about this - I'll reopen a smaller PR for /library only.

**Pull request closed without merging**